### PR TITLE
A/B test for image optimizations

### PR DESCRIPTION
**Pull request by danluu**
Related to #88786, #88925, #90741

## Proposed Changes

* A/B test impact of the above image optimization changes
* This doesn't fully back out #90741 in the experimental condition, but it does test the lazy loading part of that, which should account for almost all of the performance impact of that change
    * @sgomes, from a correctness standpoint, do see a reason just removing the lazy attribute shouldn't work?

## Why are these changes being made?

Looking retrospectively, we saw almost no impact from the changes above.

@sgomes's testing of #90741 alone found that when testing with Chrome dev tools "Fast 3G", LCP improved from 49s to 5.4s. If we add in the other two PRs, which reduce the page weight from ~50MB to ~5MB, it seems a bit surprising that we appeared to see a small impact (perhaps a 1% improvement in conversions at best, and possibly much less) given that our RUM stats show that almost 20% of users at the design picker were on a 3G or slower connection. One caveat about that stat is that Chrome appears to report 3G or slower if either latency or bandwidth are worse than some threshold. If users have high bandwidth and high latency, they could see a much smaller performance improvement than we saw in tests.

The hope is that this A/B test gives us some more info about the impact of these changes.

## Testing Instructions

We should look at the experimental and control conditions to see that we're actually displaying the correct images in the correct order in both conditions. 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by sgomes** [comment]
> * @sgomes, from a correctness standpoint, do see a reason just removing the lazy attribute shouldn't work?

@danluu There will be some differences, in that the image loading will be triggered by the DOM (instead of by JS like it was originally), which may result in the browser being able to perform some optimisations; e.g. it may prioritise images closer to the top of the page when fetching, or it may fetch them in smaller batches. This prioritisation doesn't happen in the JS version, because the fetches start when the React render happens, before anything gets committed to the DOM, so the browser doesn't have any context.

I can't estimate what the impact of what the above will be, since I'm not closely familiar with the heuristics browsers use on eagerly loaded images.

Another difference is that in the original version it's possible that some browsers might start two fetches for an image, given that it's loaded both through JS and the DOM, and they may not deduplicate pending / ongoing requests. This is hard to experimentally verify, as we'd need to test all sorts of timing conditions. Browsers should deduplicate correctly if the image is already in cache, however.

In short: while I think this is a worthwhile experiment, it's not testing exactly what we had before, and I'll leave it to your judgement as to whether it's a close enough match!

**Comment by danluu** [comment]
> There will be some differences, in that the image loading will be triggered by the DOM (instead of by JS like it was originally), which may result in the browser being able to perform some optimisations; e.g. it may prioritise images closer to the top of the page when fetching, or it may fetch them in smaller batches. This prioritisation doesn't happen in the JS version, because the fetches start when the React render happens, before anything gets committed to the DOM, so the browser doesn't have any context.

Intuitively, I would expect this to make a big difference, but when I measured the treatment arm of this experiment, I observed the same behavior I saw before all the changes under discussion were shipped, where images load in what appears to be a basically random order, with below the fold images often loading before above the fold images. In the current prod version, `loading="lazy"` prevents most of the below the fold images from loading at all, but with that gone, we don't seem to get the load order I expect to see on websites, where images that are higher up load before images that are lower. This is total speculation, but I wonder if this has to do with each `DesignPreviewImage` being created in a psuedo-random order according to whatever React "decides" to do first?

That was the big change I would be concerned about, and it seems ok. I think there will be some impact from avoiding a double image load and also from avoiding re-fetching images (which will generally be served from cache) when you scroll, but I think those should be a lot smaller than the captured changes.

**Comment by danluu** [comment]
Hmm, I checked this again but with a prod build instead of a dev build, and then the images mostly loaded in the correct order in the treatment condition, so it seems like the JS needs to be swapped out after all since. I think the main differences between the prod and deb builds should be on the JS side of things, which is apparently enough to get the images to load in the correct order or not? It's also possible there's some randomness in play, but when I checked on the prod build, only 1 or 2 below the fold images loaded before the above the fold images, which would very unlikely to happen randomly given the total number of images.

**Comment by danluu** [comment]
On testing a version with the MShots loading JS changed and not just the lazy attribute, loading the relatively small "Business" tab of the design picker took ~450s instead of ~120s and the images loaded in random order as expected on an M3 Max Macbook with "Fast 3G" set in Chrome Dev tools. I only ran a single test because these tests take so long, so the actual time may vary significantly, but seeing basically random vs. basically linear load order seems like it's very likely to reproduce and cause a large observed performance difference.
